### ISSUE TITLE
M-x shell helm-pattern is always empty

### ISSUE THREAD

**Issue by Ambrevar** [emacs-helm/helm-fish-completion#1]
Recipe:

- `M-x shell RET`
- `gi TAB` (with TAB bound to `helm-fish-completion`

Helm appears with the right candidates but the Helm pattern is empty at first.
In Eshell, the Helm pattern is properly set to `gi`.

@thierryvolpiatto Any clue what I'm missing here?

The important part of the code is

```elisp
(defun helm-fish-completion-shell-complete ()
  "Complete `shell' or `eshell' prompt with `fish-completion-complete'.
If we are in a remote location, use the old completion function instead,
since we rely on a local fish instance to suggest the completions."
  (unless (file-remote-p default-directory)
    (with-helm-current-buffer
      (helm-fish-completion-complete (buffer-substring-no-properties
                                      (save-excursion (if (eq major-mode 'shell-mode)
                                                          (comint-bol)
                                                        (eshell-bol))
                                                      (point))
                                      (point))))))
(defvar helm-fish-completion-source
  (helm-build-sync-source "Completion"
    :candidates #'helm-fish-completion-shell-complete
    :action helm-fish-completion-actions
    :keymap helm-fish-completion-map
    :fuzzy-match helm-eshell-fuzzy-match
    :persistent-action 'ignore))

;;;###autoload
(defun helm-fish-completion ()
  "Helm interface for fish completion.
This is mostly useful for `M-x shell'.
For Eshell, see `helm-fish-completion-make-eshell-source'."
  (interactive)
  (helm :sources 'helm-fish-completion-source
        :buffer "*helm-fish-completion*"))

```

**Comment by Ambrevar** [comment]
@thierryvolpiatto Friendly ping :)  Any idea?

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> Recipe:
>
>   • M-x shell RET
>   • gi TAB (with TAB bound to helm-fish-completion
>
> Helm appears with the right completions but the Helm pattern is empty at first.
> In Eshell, the Helm pattern is properly set to gi.

Use :input ?

-- 
Thierry

Get my Gnupg key:
gpg --keyserver pgp.mit.edu --recv-keys 59F29997

**Comment by Ambrevar** [comment]
Of course! :)

Ouch, my Helm programming skills are getting rusty... :p

Fixed with 3c3515fb72f808235b0e457b569f21e686bbc6bc,
thanks!
